src/main/java/org/opensearch/neuralsearch/search/collector/SimpleFieldCollector.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.collector;
+
+import java.io.IOException;
+import java.util.Objects;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.LeafCollector;
+import org.apache.lucene.search.Sort;
+import org.opensearch.neuralsearch.search.HitsThresholdChecker;
+
+/*
+  SimpleFieldCollector collects the sorted results at the shard level for every individual query.
+  It collects the list of TopFieldDocs.
+ */
+public final class SimpleFieldCollector extends HybridTopFieldDocSortCollector {
+
+    public SimpleFieldCollector(int numHits, HitsThresholdChecker hitsThresholdChecker, Sort sort) {
+        super(numHits, hitsThresholdChecker, sort);
+    }
+
+    @Override
+    public LeafCollector getLeafCollector(LeafReaderContext context) {
+        docBase = context.docBase;
+
+        return new HybridTopDocSortLeafCollector(null) {
```

Comment:
this looks weird, it should be possible to define no args constructor that passes param as `null` at parent level (HybridTopDocSortLeafCollector) extend it at the child level (SimpleFieldCollector).

Reply:
ack